homeworks/03-data-structure/mergeSort.ts

```diff
@@ -0,0 +1,53 @@
+export enum CompareResult {
+  EQ = 0,
+  GT = 1,
+  LS = -1
+}
+
+export type Comparator<T> = (a: T, b: T) => CompareResult;
+
+export function mergeSort<V>(xs: V[], comparator: Comparator<V>): V[] {
+  const copy = xs.concat();
+  if (copy.length <= 1) {
+    return copy
+  }
+  mergeSortRec(copy, comparator, 0, xs.length - 1);
+  return copy;
+}
+
+function mergeSortRec<V>(xs: V[], comparator: Comparator<V>, left: number, right: number) {
+  if (left >= right) {
+    return;
+  }
+  const middle = ((right + left) / 2) | 0;
+  mergeSortRec(xs, comparator, left, middle)
+  mergeSortRec(xs, comparator, middle + 1, right);
+  merge(xs, comparator, left, middle, right);
+}
+
+function merge<V>(xs: V[], comparator: Comparator<V>, left: number, middle: number, right: number) {
+  const ls = [];
```

Comment:
What is `ls`, `xs`, `rs`, etc.? It's extremely unreadable. Use readable and understandable naming and don't need to shorten.